Lusas_Adapter/CRUD/Create/Elements/Surface.cs

```diff
@@ -70,7 +74,39 @@ private IFSurface CreateSurface(Panel panel)
 
             IFSurface lusasSurface = d_LusasData.createSurfaceBy(edges.ToArray());
 
-            if (lusasSurface != null)
+            List<IFSurface> openings = new List<IFSurface>();
+
+            foreach (Opening opening in panel.Openings)
+               {
+                string openingID = GetAdapterId<string>(opening);
+
```

Comment:
In here can we do a check the `Opening` does not intersect with the `Panel`, i.e. you can make a `Panel` with an `Opening` that shares an `Edge` partially or in full. The `Create` method tries to resolve it, but you cannot push to Lusas. So we should just raise an error here.

Reply:
Added a private method for EdgeIntersection to catch this.